ApiBundle/Controller/TrashcanController.php

```diff
@@ -58,12 +58,42 @@ public function restoreAction($trashItemId)
         $trashItem = $this->get('open_orchestra_model.repository.trash_item')->find($trashItemId);
         /* @var $entity SoftDeleteableInterface */
         $entity = $trashItem->getEntity();
-        $dm = $this->get('object_manager');
+        $om = $this->get('object_manager');
 
         if ($this->isValid($entity, 'restore')) {
             $this->get('open_orchestra_backoffice.restore_entity.manager')->restore($entity);
-            $dm->remove($trashItem);
-            $dm->flush();
+            $om->remove($trashItem);
+            $om->flush();
+
+            return array();
+        }
+
+        return $this->getViolations();
+    }
+
+
+    /**
+     * @param $trashItemId
+     *
+     * @Config\Route("/{trashItemId}/delete", name="open_orchestra_api_trashcan_delete")
+     * @Config\Method({"DELETE"})
+
+     * @Config\Security("is_granted('ROLE_ACCESS_DELETED_TRASHCAN')")
+     *
+     * @return array|mixed
+     */
+    public function deleteAction($trashItemId)
```

Comment:
Maybe more a ``removeAction`` with a ``/remove`` url to mark the difference between soft delete and hard delete.